src/main/java/com/puppycrawl/tools/checkstyle/checks/TrailingCommentCheck.java

```diff
@@ -261,15 +263,19 @@ private void checkBlockComment(DetailAST ast) {
         final DetailAST firstChild = ast.getFirstChild();
         final DetailAST lastChild = ast.getLastChild();
         final String comment = firstChild.getText();
-        String line = getLines()[lineNo - 1];
+        int[] lineCodePoints = getLineCodePoints(lineNo - 1);
 
-        if (line.length() > lastChild.getColumnNo() + 1) {
-            line = line.substring(lastChild.getColumnNo() + 2);
+        if (lineCodePoints.length > lastChild.getColumnNo() + 1) {
+            lineCodePoints = Arrays.copyOfRange(lineCodePoints,
+                    lastChild.getColumnNo() + 2, lineCodePoints.length);
         }
 
+        String line = new String(lineCodePoints, 0, lineCodePoints.length);
         line = FORMAT_LINE.matcher(line).replaceAll("");
 
-        final String lineBefore = getLines()[lineNo - 1].substring(0, ast.getColumnNo());
+        final int[] lineBeforeCodePoints =
+                Arrays.copyOfRange(getLineCodePoints(lineNo - 1), 0, ast.getColumnNo());
+        final String lineBefore = new String(lineBeforeCodePoints, 0, lineBeforeCodePoints.length);
```

Comment:
I think it may make more sense here to just use `getLine`, @MUzairS15 what do you think?

Reply:
When we take `substring`, due to the behaviour of codepoints wrong string is being matched.
While using this,  
`final int[] lineBeforeCodePoints = Arrays.copyOfRange(getLineCodePoints(lineNo - 1), 0, ast.getColumnNo());` 
and converting to string afterwards makes sure correct part of string is captured and matched.

Reply:
Resolved, thanks for explanation.